### PR TITLE
Update react-native to version 0.55.4

### PR DESCRIPTION
**Pull request by IvanSakharovskii**
Updated libraries for iOS and android, refactored JS code with new react-native standards.

### CODE REVIEW

File: sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinLayoutController.java [hunk 1]
```diff
@@ -199,16 +199,14 @@ private void initializeSkin(Application app, OoyalaSkinLayout l, OoyalaPlayer p,
     _reactInstanceManager = ReactInstanceManager.builder()
         .setApplication(app)
         .setBundleAssetName(skinOptions.getBundleAssetName())
-        .setJSMainModuleName("index.android")
+        .setJSBundleFile("assets://index.android.jsbundle")
+//        .setJSMainModuleName("index.android")
```
Comment (brol1dev) [inline sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinLayoutController.java:203]: let's remove this line if it's not needed anymore.

File: sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinPlayerObserver.java [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import java.util.Map;
 import java.util.Observable;
 import java.util.Observer;
+import java.util.Set;
```
Comment (brol1dev) [inline sdk/android/skin/src/main/java/com/ooyala/android/skin/OoyalaSkinPlayerObserver.java:19]: are you using a Set in here? If not remove this import

**Review by brol1dev — CHANGES_REQUESTED** [review]
Awesome refactor. Thanks!

Just added a couple of comments. Also want to ask if you tested building the SkinSDK and using it?

**Review by akshaydch11 — APPROVED** [review]
same comments as Eric added. Rest looks good

**Comment by akshaydch11** [comment]
I am merging this PR as is now.
@IvanSakharovskii  Please add commented changes in separate PR. 
Also we need PR to update dependencies for all sample apps which uses skin.

**Pull request merged**